testing/src/scenario/_ops_main_mock.py

```diff
@@ -270,6 +272,28 @@ def _close(self):
         self.state = dataclasses.replace(self.state, deferred=deferred, stored_states=stored_state)
 
     def _destroy(self):
+        # Must be run first, so that ops._main trace span is finished.
         super()._destroy()
         if self._tracing_mock:
+            assert self._tracing_exporter
+            data = self._tracing_exporter.get_finished_spans()
+            # FIXME: think if this is worth the effort:
+            # OpenTelemetry SDK stores spans chronologically by end time.
+            # This means that `ops._main` span is last.
+            #
+            # We'd want to offer trace data in a consistent order,
+            # ideally chronologically by start_time.
+            # However, a unit test could be run with time frozen.
+            #
+            # `reversed` ensures that outermost spans appear earlier...
+            # but inverts the order of consecutive spans...
+
+            # Minor discussion: what it takes to type-check cleanly.
+            #
+            # .start_time is always set, but it's declared nullable.
+            self.trace_data = sorted(reversed(data), key=lambda s: s.start_time or 0)
+            from operator import attrgetter
+
+            # this option type checks cleanly though
+            self.trace_data = sorted(reversed(data), key=attrgetter('start_time'))
```

Comment:
I've always personally quite liked using the `operator` methods, and I think here avoiding the 'or 0' is nice.

I do get the point about the `lambda` version being understandable without needing to know about the `operator` module, though. So I'm slightly in favour of the latter, but don't object to the former.

Reply:
This is no longer relevant, as I've decided not to order the spans.